### PR TITLE
refactor: harden serialized bounds checks

### PR DESCRIPTION
**Pull request by sjinks**
## Summary

Hardens serialized string parsing to prevent access to malformed input out of bounds while preserving malformed serialized segments unchanged through the existing fallback path.

## Changes

- Uses safer serialized content start handling and adds explicit bounds checks before escaped-byte and terminator lookahead reads.
- Returns malformed-data errors when parsing cannot safely continue, including incomplete escape pairs, incomplete terminator probes, and missing terminators.
- Handles invalid or overflowing declared byte counts explicitly by returning an error from declared-size parsing rather than relying on implicit loop termination.
- Expands regression coverage for malformed/truncated serialized inputs, including:
  - prefix-only and unterminated content
  - trailing incomplete escapes and EOF probe edges
  - oversized and overflowed declared byte counts
  - malformed helper-input edge cases for unescaping behavior

## Testing and Validation

- Local validation:
  - go test ./...
- CI validation:
  - Build and Test workflow check build: SUCCESS

## Risks and Follow-up

- Residual risk is low and centered on uncommon malformed serialized payload shapes not represented in existing fixtures.
- Future hardening could add fuzz coverage around serialized parser boundary conditions.

Related issue: https://linear.app/a8c/issue/PLTFRM-2237/out-of-bounds-index-access

### CODE REVIEW

File: searchreplace/search-replace.go [hunk 1]
```diff
@@ -85,11 +85,7 @@ func fixLineWithSerializedData(linePart []byte, replacements []*Replacement) (*S
 
 	originalByteSize, _ := strconv.Atoi(string(originalBytes))
```
[the remark this answers is not in the document]
Comment (sjinks) [reply searchreplace/search-replace.go]: Thanks — addressed in ce7f1ce. `strconv.Atoi` is now error-checked and returns a `faulty serialized data: invalid declared byte count` error so the existing `FixLine` fallback path keeps the malformed bytes untouched, matching the other malformed-data branches in this function.

One small correction on the failure mode you described: for the overflow case `Atoi` actually returns `math.MaxInt` together with `strconv.ErrRange` (not `0`), so the previous code still fell back via the end-of-loop "end of serialized data not found" path rather than rewriting the prefix — the existing `overflowing declared byte count` test already exercised that. The change in this commit just makes the intent explicit and the fallback direct instead of relying on the loop termination. Added an extra regression case (`negative-ish overflowing declared byte count via even-larger digits`) that exercises the new explicit-error path. `go test ./...` is green.

**Review by rebeccahum — APPROVED** [review]
